### PR TITLE
Allow to manage article autopublishing feature by rules

### PR DESCRIPTION
**Pull request by takeit**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | no
| Deprecations?             | no
| Tests pass?               | yes|no
| Changelog update required | yes
| Fixed tickets             | [SWP-305](https://dev.sourcefabric.org/browse/SWP-305)
| License                   | AGPLv3

TODO:

- [x] - functional tests
- [x] - fix specs
- [x] - documentation

### CODE REVIEW

File: src/SWP/Bundle/ContentBundle/Rule/Applicator/ArticleRuleApplicator.php [hunk 1]
```diff
@@ -77,6 +88,10 @@ public function apply(RuleInterface $rule, RuleSubjectInterface $subject)
 
         $subject->setTemplateName($configuration[$this->supportedKeys[1]]);
 
+        if ((bool) $configuration[$this->supportedKeys[2]]) {
```
Comment (ahilles107) [inline src/SWP/Bundle/ContentBundle/Rule/Applicator/ArticleRuleApplicator.php]: I don't like relying on array index. Why not `if (in_array('published', $configuration) && true === $configuration['published'])`

Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Rule/Applicator/ArticleRuleApplicator.php]: there is option resolver which resolves the configuration array before it goes here and it makes sure that this key is always here.

Comment (ahilles107) [inline src/SWP/Bundle/ContentBundle/Rule/Applicator/ArticleRuleApplicator.php]: yeah, but when you add new element at start of array then it's hard to catch that bug.

Comment (ahilles107) [inline src/SWP/Bundle/ContentBundle/Rule/Applicator/ArticleRuleApplicator.php]: constant sound better there for me than keys in array ;)

Comment (takeit) [inline src/SWP/Bundle/ContentBundle/Rule/Applicator/ArticleRuleApplicator.php]: tests will fail and tell you whats wrong anyway, but yeah I agree referring by key is less readable here tho, will change that

**Pull request merged**